### ISSUE TITLE
Backspace in text field causes error 

### ISSUE THREAD

**Issue by domesticmouse** [flutter/flutter#116953]
Code: https://github.com/domesticmouse/codelabs/tree/ffigen-step07/ffigen_codelab/step_07
To be published as https://github.com/flutter/codelabs/tree/main/ffigen_codelab/step_07 once https://github.com/flutter/codelabs/pull/1201 lands

Entering some text then backspacing causes the following error after a couple of edits:

```console
Launching lib\main.dart on Windows in debug mode...
lib\main.dart:1
Connecting to VM Service at ws://127.0.0.1:51972/dExHVgtAaZs=/ws
Unable to parse JSON message:
The document is empty.
Unable to parse JSON message:
The document is empty.

════════ Exception caught by services library ══════════════════════════════════
The following assertion was thrown during a platform message callback:
A KeyRepeatEvent is dispatched, but the state shows that the physical key is not pressed. If this occurs in real application, please report this bug to Flutter. If this occurs in unit tests, please ensure that simulated events follow Flutter's event model as documented in `HardwareKeyboard`. This was the event: KeyRepeatEvent#904f7(physicalKey: PhysicalKeyboardKey#700e1(usbHidUsage: "0x000700e1", debugName: "Shift Left"), logicalKey: LogicalKeyboardKey#00102(keyId: "0x200000102", keyLabel: "Shift Left", debugName: "Shift Left"), character: null, timeStamp: 0:36:17.101642)
'package:flutter/src/services/hardware_keyboard.dart':
package:flutter/…/services/hardware_keyboard.dart:1
Failed assertion: line 432 pos 16: '_pressedKeys.containsKey(event.physicalKey)'

2

Either the assertion indicates an error in the framework itself, or we should provide substantially more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=2_bug.md

When the exception was thrown, this was the stack
#2      HardwareKeyboard._assertEventIsRegular.<anonymous closure>
package:flutter/…/services/hardware_keyboard.dart:432
#3      HardwareKeyboard._assertEventIsRegular
package:flutter/…/services/hardware_keyboard.dart:443
#4      HardwareKeyboard.handleKeyEvent
package:flutter/…/services/hardware_keyboard.dart:535
#5      KeyEventManager.handleRawKeyMessage
package:flutter/…/services/hardware_keyboard.dart:881
#6      BasicMessageChannel.setMessageHandler.<anonymous closure>
package:flutter/…/services/platform_channel.dart:197
#7      _DefaultBinaryMessenger.setMessageHandler.<anonymous closure>
package:flutter/…/services/binding.dart:387
#8      _invoke2 (dart:ui/hooks.dart:186:13)
#9      _ChannelCallbackRecord.invoke (dart:ui/channel_buffers.dart:42:5)
#10     _Channel.push (dart:ui/channel_buffers.dart:132:31)
#11     ChannelBuffers.push (dart:ui/channel_buffers.dart:329:17)
#12     PlatformDispatcher._dispatchPlatformMessage (dart:ui/platform_dispatcher.dart:599:22)
#13     _dispatchPlatformMessage (dart:ui/hooks.dart:89:31)
(elided 2 frames from class _AssertionError)
════════════════════════════════════════════════════════════════════════════════
Unable to parse JSON message:
The document is empty.

════════ Exception caught by services library ══════════════════════════════════
A KeyRepeatEvent is dispatched, but the state shows that the physical key is not pressed. If this occurs in real application, please report this bug to Flutter. If this occurs in unit tests, please ensure that simulated events follow Flutter's event model as documented in `HardwareKeyboard`. This was the event: KeyRepeatEvent#904f7(physicalKey: PhysicalKeyboardKey#700e1(usbHidUsage: "0x000700e1", debugName: "Shift Left"), logicalKey: LogicalKeyboardKey#00102(keyId: "0x200000102", keyLabel: "Shift Left", debugName: "Shift Left"), character: null, timeStamp: 0:36:17.101642)
'package:flutter/src/services/hardware_keyboard.dart':
package:flutter/…/services/hardware_keyboard.dart:1
Failed assertion: line 432 pos 16: '_pressedKeys.containsKey(event.physicalKey)'

════════════════════════════════════════════════════════════════════════════════
```

Obligatory debug info:

```console
PS C:\Users\brett> flutter doctor -v
[✓] Flutter (Channel stable, 3.3.9, on Microsoft Windows [Version 10.0.22621.900], locale en-AU)
    • Flutter version 3.3.9 on channel stable at C:\Users\brett\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b8f7f1f986 (3 weeks ago), 2022-11-23 06:43:51 +0900
    • Engine revision 8f2221fbef
    • Dart version 2.18.5
    • DevTools version 2.15.0

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2022 17.0.5)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.0.32112.339
    • Windows 10 SDK version 10.0.19041.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).

[✓] IntelliJ IDEA Community Edition (version 2022.1)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2022.1
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.74.0)
    • VS Code at C:\Users\brett\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.54.0

[✓] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22621.900]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 108.0.5359.99
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 107.0.1418.42

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.
```

**Comment by loic-sharma** [comment]
I wasn't able to reproduce this using Flutter stable channel v3.3.9. @dkwingsmt Do you know what may be causing this?

<details>
<summary>My `flutter doctor -v` output</summary>

I'm on same Flutter build, similar Windows version, and same Australian locale.

```
[✓] Flutter (Channel stable, 3.3.9, on Microsoft Windows [Version 10.0.22621.819], locale en-AU)
    • Flutter version 3.3.9 on channel stable at C:\Code\f\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b8f7f1f986 (3 weeks ago), 2022-11-23 06:43:51 +0900
    • Engine revision 8f2221fbef
    • Dart version 2.18.5
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at C:\Users\sharm\AppData\Local\Android\sdk
    • Platform android-33, build-tools 32.1.0-rc1
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2022 17.2.5)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.2.32616.157
    • Windows 10 SDK version 10.0.22000.0

[✓] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[✓] VS Code (version 1.74.0)
    • VS Code at C:\Users\sharm\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.54.0

[✓] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22621.819]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 107.0.5304.88
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 108.0.1462.46

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

**Comment by dkwingsmt** [comment]
So a Shift repeat event is dispatched when Shift is recorded as not pressed. I'm wondering if it's a synthesization from modifier states caused the release of the key.

**Comment by dkwingsmt** [comment]
I would highly appreciate if you could find a reproduction. Otherwise you can modify the engine and print the arguments of `keyboard_manager_->HandleMessage` in `window.cc`.

**Comment by domesticmouse** [comment]
> I would highly appreciate if you could find a reproduction. Otherwise you can modify the engine and print the arguments of `keyboard_manager_->HandleMessage` in `window.cc`.

Is there docs somewhere on building the engine for windows?

**Comment by dkwingsmt** [comment]
Yeah, it's written [in the wiki](https://github.com/flutter/flutter/wiki/Setting-up-the-Engine-development-environment). But it's a tedious process. I would say let's leave this issue open and if you encounter the next time, pay attention to what you've pressed.

**Comment by JohnF17** [comment]
Hello @dkwingsmt and @domesticmouse, I've been encountering this issue way too often that I can proudly say I can help you reproduce that error with ease (sadly).

#### Steps to reproduce 
1. Create a textfield widget
2. Click on it (put it in focus)
3. Hold the shift key for about 3-5 seconds
then voila errors everywhere


#### Current behaviour (bug)
- Backspace doesn't clear characters anymore while simultaneously throwing the previous errors
- Holding the shift key prints more errors but capitalises the characters
- Although very RARE it might cause the app to crash
- Hot reload doesn't fix it
- Maybe (just maybe) hot restart doesn't fix it too, you'd have to stop the whole process


#### Expected behaviour (correct)
- Clear characters and stop showing errors


#### Errors thrown
```
Unable to parse JSON message:
The document is empty.

════════ Exception caught by services library ══════════════════════════════════
The following assertion was thrown during a platform message callback:
A KeyRepeatEvent is dispatched, but the state shows that the physical key is not pressed. If this occurs in real application, please report this bug to Flutter. If this occurs in unit tests, please ensure that simulated events follow Flutter's event model as documented in `HardwareKeyboard`. This was the event: KeyRepeatEvent#5e0fd(physicalKey: PhysicalKeyboardKey#700e5(usbHidUsage: "0x000700e5", debugName: "Shift Right"), logicalKey: LogicalKeyboardKey#00103(keyId: "0x200000103", keyLabel: "Shift Right", debugName: "Shift Right"), character: null, timeStamp: 25:11:29.787110)
'package:flutter/src/services/hardware_keyboard.dart':
package:flutter/…/services/hardware_keyboard.dart:1
Failed assertion: line 432 pos 16: '_pressedKeys.containsKey(event.physicalKey)'

Either the assertion indicates an error in the framework itself, or we should provide substantially more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=2_bug.md

When the exception was thrown, this was the stack
#2      HardwareKeyboard._assertEventIsRegular.<anonymous closure>
package:flutter/…/services/hardware_keyboard.dart:432
#3      HardwareKeyboard._assertEventIsRegular
package:flutter/…/services/hardware_keyboard.dart:443
#4      HardwareKeyboard.handleKeyEvent
package:flutter/…/services/hardware_keyboard.dart:535
#5      KeyEventManager.handleRawKeyMessage
package:flutter/…/services/hardware_keyboard.dart:881
#6      BasicMessageChannel.setMessageHandler.<anonymous closure>
package:flutter/…/services/platform_channel.dart:197
#7      _DefaultBinaryMessenger.setMessageHandler.<anonymous closure>
package:flutter/…/services/binding.dart:387
#8      _invoke2 (dart:ui/hooks.dart:186:13)
#9      _ChannelCallbackRecord.invoke (dart:ui/channel_buffers.dart:42:5)
#10     _Channel.push (dart:ui/channel_buffers.dart:132:31)
#11     ChannelBuffers.push (dart:ui/channel_buffers.dart:329:17)
#12     PlatformDispatcher._dispatchPlatformMessage (dart:ui/platform_dispatcher.dart:599:22)
#13     _dispatchPlatformMessage (dart:ui/hooks.dart:89:31)
(elided 2 frames from class _AssertionError)
```

Debug info:
```
flutter doctor -v
[√] Flutter (Channel stable, 3.3.10, on Microsoft Windows [Version 10.0.22000.1335], locale en-US)
    • Flutter version 3.3.10 on channel stable at C:\DevTools\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 135454af32 (2 weeks ago), 2022-12-15 07:36:55 -0800
    • Engine revision 3316dd8728
    • Dart version 2.18.6
    • DevTools version 2.15.0

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at C:\Users\John\AppData\Local\Android\sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.3.2)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.3.32819.101
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[√] VS Code (version 1.74.2)
    • VS Code at C:\Users\John\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.56.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22000.1335]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 108.0.5359.125
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 108.0.1462.46

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

#### Other comment
I'm not sure if it happens on other desktop platforms but I've encountered it on Windows only so far.
And I think I do have to say that this happens in debug mode (not sure if it happens on profile or release mode)
How often does this happen? **All the time** 
Does it happen with other keys?  **Yes** with the ctrl keys too (I'll put those errors after this section)
How often do crashes happen? Almost never but you could smell one coming when it lags so much
If you need any more information tag me (I don't know if it sends me notifications but worth the try)

#### Right ctrl key error
```
════════════════════════════════════════════════════════════════════════════════
Unable to parse JSON message:
The document is empty.

════════ Exception caught by services library ══════════════════════════════════
A KeyRepeatEvent is dispatched, but the state shows that the physical key is not pressed. If this occurs in real application, please report this bug to Flutter. If this occurs in unit tests, please ensure that simulated events follow Flutter's event model as documented in `HardwareKeyboard`. This was the event: KeyRepeatEvent#2c956(physicalKey: PhysicalKeyboardKey#700e4(usbHidUsage: "0x000700e4", debugName: "Control Right"), logicalKey: LogicalKeyboardKey#00101(keyId: "0x200000101", keyLabel: "Control Right", debugName: "Control Right"), character: null, timeStamp: 26:07:12.139183)
'package:flutter/src/services/hardware_keyboard.dart':
package:flutter/…/services/hardware_keyboard.dart:1
Failed assertion: line 432 pos 16: '_pressedKeys.containsKey(event.physicalKey)'

════════════════════════════════════════════════════════════════════════════════
```

/label ~Bug

**Comment by loic-sharma** [comment]
@JohnF17 I can reproduce your issue on Flutter's `stable` channel, but it works as expected on the [`master` channel](https://docs.flutter.dev/development/tools/sdk/upgrading#switching-flutter-channels). 

Could you try reproducing your issue on the `master` channel? You can try this using:

```
flutter channel master
flutter upgrade
```

**Comment by JohnF17** [comment]
I don't know if its a noob question but Is the master channel safe to use 😅?
Currently working on a huge project, really wouldn't want to risk it if its irreversible but if you give me the go-ahead I'll do what you recommended and let you know about it right now.

**Comment by loic-sharma** [comment]
@JohnF17 No worries, it's a valid question! For your case, I'd recommend switching to `master` temporarily just to test this bug. Once your test is over, I'd switch back to `stable`. FYI the `master` is Flutter's bleeding edge: we expect it to be high quality, but it you might occasionally find bugs. You can find out more here: https://github.com/flutter/flutter/wiki/Flutter-build-release-channels

**Comment by JohnF17** [comment]
@loic-sharma If it was up to me i would have replied much sooner but the channel switching, channel upgrading, flutter initializing, pub upgrading and launching the app is what took this long 😅 

`
[√] Flutter (Channel master, 3.7.0-15.0.pre.11, on Microsoft Windows [Version 10.0.22000.1335], locale en-US)
    • Flutter version 3.7.0-15.0.pre.11 on channel master at C:\DevTools\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 507062032f (20 hours ago), 2023-01-06 01:13:18 +0100
    • Engine revision 5bd90d6e77
    • Dart version 3.0.0 (build 3.0.0-85.0.dev)
    • DevTools version 2.20.0

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[√] VS Code (version 1.74.2)
    • VS Code at C:\Users\John\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.56.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22000.1335]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 108.0.5359.125
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 108.0.1462.46

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
`
But I'm running into other issues now 😓

Errors:
`
Running "flutter pub get" in flutter_tools...
Resolving dependencies in C:\DevTools\flutter\packages\flutter_tools...
  archive 3.3.2 (3.3.5 available)
  intl 0.17.0 (0.18.0 available)
Got dependencies in C:\DevTools\flutter\packages\flutter_tools!
Downloading windows-x64/windows-x64-flutter tools...
Downloading windows-x64/flutter-cpp-client-wrapper tools...
Downloading windows-x64-profile/windows-x64-flutter tools...
Downloading windows-x64-release/windows-x64-flutter tools...
Launching lib\main.dart on Windows in debug mode...
lib\main.dart:1
Nuget.exe not found, trying to download or use cached version.
/C:/Users/John/AppData/Local/Pub/Cache/hosted/pub.dev/sortedmap-0.5.2/lib/src/treeset.dart(17,3): error G5FE39F1E: Type 'BidirectionalIterator' not found. [P:\Flutter\Full Stack\HA\hilcoe_desktop\build\windows\flutter\flutter_assemble.vcxproj]
/C:/Users/John/AppData/Local/Pub/Cache/hosted/pub.dev/sortedmap-0.5.2/lib/src/treeset.dart(21,3): error G5FE39F1E: Type 'BidirectionalIterator' not found. [P:\Flutter\Full Stack\HA\hilcoe_desktop\build\windows\flutter\flutter_assemble.vcxproj]
/C:/Users/John/AppData/Local/Pub/Cache/hosted/pub.dev/sortedmap-0.5.2/lib/src/treeset.dart(23,3): error G5FE39F1E: Type 'BidirectionalIterator' not found. [P:\Flutter\Full Stack\HA\hilcoe_desktop\build\windows\flutter\flutter_assemble.vcxproj]
/C:/Users/John/AppData/Local/Pub/Cache/hosted/pub.dev/sortedmap-0.5.2/lib/src/treeset.dart(184,3): error G5FE39F1E: Type 'BidirectionalIterator' not found. [P:\Flutter\Full Stack\HA\hilcoe_desktop\build\windows\flutter\flutter_assemble.vcxproj]
/C:/Users/John/AppData/Local/Pub/Cache/hosted/pub.dev/sortedmap-0.5.2/lib/src/treeset.dart(187,3): error G5FE39F1E: Type 'BidirectionalIterator' not found. [P:\Flutter\Full Stack\HA\hilcoe_desktop\build\windows\flutter\flutter_assemble.vcxproj]
/C:/Users/John/AppData/Local/Pub/Cache/hosted/pub.dev/sortedmap-0.5.2/lib/src/treeset.dart(305,3): error G5FE39F1E: Type 'BidirectionalIterator' not found. [P:\Flutter\Full Stack\HA\hilcoe_desktop\build\windows\flutter\flutter_assemble.vcxproj]
/C:/Users/John/AppData/Local/Pub/Cache/hosted/pub.dev/sortedmap-0.5.2/lib/src/treeset.dart(527,31): error G5FE39F1E: Type 'BidirectionalIterator' not found. [P:\Flutter\Full Stack\HA\hilcoe_desktop\build\windows\flutter\flutter_assemble.vcxproj]
/C:/Users/John/AppData/Local/Pub/Cache/hosted/pub.dev/sortedmap-0.5.2/lib/src/treeset.dart(579,29): error G5FE39F1E: Type 'BidirectionalIterator' not found. [P:\Flutter\Full Stack\HA\hilcoe_desktop\build\windows\flutter\flutter_assemble.vcxproj]

/C:/Users/John/AppData/Local/Pub/Cache/hosted/pub.dev/fluent_ui-4.1.3/lib/src/controls/inputs/slider.dart(363,8): error GE5CFE876: The method '_CustomTrackShape.paint' has fewer named arguments than those of overridden method 'RoundedRectSliderTrackShape.paint'. [P:\Flutter\Full Stack\HA\hilcoe_desktop\build\windows\flutter\flutter_assemble.vcxproj]
/C:/Users/John/AppData/Local/Pub/Cache/hosted/pub.dev/fluent_ui-4.1.3/lib/src/controls/inputs/slider.dart(363,8): error GE5CFE876: The method '_CustomTrackShape.paint' doesn't have the named parameter 'secondaryOffset' of overridden method 'RoundedRectSliderTrackShape.paint'. [P:\Flutter\Full Stack\HA\hilcoe_desktop\build\windows\flutter\flutter_assemble.vcxproj]
/C:/Users/John/AppData/Local/Pub/Cache/hosted/pub.dev/fluent_ui-4.1.3/lib/src/controls/form/text_box.dart(29,35): error G4906CC7E: The parameter 'details' of the method '_TextBoxSelectionGestureDetectorBuilder.onSingleTapUp' has type 'TapUpDetails', which does not match the corresponding type, 'TapDragUpDetails', in the overridden method, 'TextSelectionGestureDetectorBuilder.onSingleTapUp'. [P:\Flutter\Full Stack\HA\hilcoe_desktop\build\windows\flutter\flutter_assemble.vcxproj]
/C:/Users/John/AppData/Local/Pub/Cache/hosted/pub.dev/fluent_ui-4.1.3/lib/src/controls/form/text_box.dart(50,42): error G4906CC7E: The parameter 'details' of the method '_TextBoxSelectionGestureDetectorBuilder.onDragSelectionEnd' has type 'DragEndDetails', which does not match the corresponding type, 'TapDragEndDetails', in the overridden method, 'TextSelectionGestureDetectorBuilder.onDragSelectionEnd'. [P:\Flutter\Full Stack\HA\hilcoe_desktop\build\windows\flutter\flutter_assemble.vcxproj]
/C:/Users/John/AppData/Local/Pub/Cache/hosted/pub.dev/fluent_ui-4.1.3/lib/src/controls/form/text_box.dart(44,25): error GC2F972A8: The argument type 'TapUpDetails' can't be assigned to the parameter type 'TapDragUpDetails'. [P:\Flutter\Full Stack\HA\hilcoe_desktop\build\windows\flutter\flutter_assemble.vcxproj]
/C:/Users/John/AppData/Local/Pub/Cache/hosted/pub.dev/sortedmap-0.5.2/lib/src/treeset.dart(254,36): error GD3C8AF11: 'BidirectionalIterator' isn't a type. [P:\Flutter\Full Stack\HA\hilcoe_desktop\build\windows\flutter\flutter_assemble.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(247,5): error MSB8066: Custom build for 'P:\Flutter\Full Stack\HA\hilcoe_desktop\build\windows\CMakeFiles\fc0786c1245c88014e7cc24aa9f115b6\flutter_windows.dll.rule;P:\Flutter\Full Stack\HA\hilcoe_desktop\build\windows\CMakeFiles\3e5cfa731716647248795c9489cb938c\flutter_assemble.rule' exited with code 1. [P:\Flutter\Full Stack\HA\hilcoe_desktop\build\windows\flutter\flutter_assemble.vcxproj]
Exception: Build process failed.
`

**Comment by loic-sharma** [comment]
From those errors, it seems like your dependencies fluent_ui and sortedmap don't work on the latest Flutter version. They may need updates from their respective maintainers. Could you test if `TextField`w works as expected on a new app?

**Comment by JohnF17** [comment]
@loic-sharma  Works as expected, glad to say the issue can not reproduced 🎊, although the problem is ... if I do switch back to stable channel the unresponsiveness along with its error feed would persist, and if I stay on this channel .. well.. the app wouldn't run. I know all of you are working really hard to make this framework as friendly and as error free as possible and I must say I appreciate the hard (/smart) work and commitment you all have towards this project, although I do have to ask with this error currently lurking about, is it safe to publish a production app in stable channel (having in mind I can't not use Fluent_UI  as its rooted deep in to the look and feel of the app) and would you put a time limit on when this `P3` issue might be fixed?

**Comment by loic-sharma** [comment]
> would you put a time limit on when this `P3` issue might be fixed?

It'll be fixed as part of the next release. Sadly, the next release date hasn't been announced yet. 

I'd consider trying Flutter's `beta` channel, this contains the changes that will become the next `stable`.

> is it safe to publish a production app in stable channel

Google has shipped Windows Flutter apps like [Google VPN](https://twitter.com/timsneath/status/1590449925300846592?s=46&t=La7OuuZhLwP9EolbvGi_CA). That said, at the end of the day that's a decision you should make for yourself 😄

**Comment by JohnF17** [comment]
> It'll be fixed as part of the next release.

That's great news, i'm glad the error has been acknowledged and has/will be provided a fix for. 

> I'd consider trying Flutter's `beta` channel, this contains the changes that will become the next `stable`.

Hmm so if I'm not mistaken, master is the volatile one while providing all the latest features, beta is semi-volatile and is the middle ground (or closer to stable), then there's stable which is firm with fewer ground breaking updates.

> at the end of the day that's a decision you should make for yourself 😄

Indeed, Let's hope I make the right one

**Comment by domesticmouse** [comment]
I'd more characterise `beta` as `stable + 1`. You will find Flutter's `master` channel is more stable than `master` branches in most projects. Google's internal Flutter projects all run on `master`, for example.

**Comment by JohnF17** [comment]
> You will find Flutter's `master` channel is more stable than `master` branches in most projects. Google's internal Flutter projects all run on `master`, for example.

@domesticmouse Ah really wish I had known that before using dependencies that only work well on `stable` apparently 😓. Seems the master channel is the way to go for the next ones although i do feel i'm missing out on great changes here for this project 😞.

**Comment by domesticmouse** [comment]
The wider Flutter community are mainly on stable. On the upside they'll be migrating to the current beta soon enough. You are just going there sooner than most. 🤓

**Comment by JohnF17** [comment]
> On the upside they'll be migrating to the current beta soon enough.

@domesticmouse Can't wait, although i do have to clarify 😅 i haven't checked if the error's still present on the `beta` channel might as well try to reproduce the error and let y'all know before its stable. I'll be commenting soon after ...

**Comment by domesticmouse** [comment]
I keep CI running on both `beta` and `stable` for the repos I'm responsible for. It's easier knowing what's about to land in stable. 😎

**Comment by JohnF17** [comment]
> might as well try to reproduce the error
> I keep CI running on both `beta` and `stable` for the repos I'm responsible for.

@domesticmouse I might as well not then 😁, seems like y'all got it covered.

> I keep CI running on both `beta` and `stable` for the repos I'm responsible for.

Off topic here but Can I just say that is so cool, I aspire to be a dev like you guys but for some, circumstances don't always let us be what we know we can become, but hey we're always gonna keep trying 💪. Don't want to take your time, Thanks for all the good work y'all put in, stay safe.

**Comment by domesticmouse** [comment]
Please try your code on `beta`, it's useful feedback for our team here.

**Comment by JohnF17** [comment]
Just realized I miss interpreted it, my bad 😅.
I'll report back soon ...

**Comment by JohnF17** [comment]
@domesticmouse Good news ... works as expected, the error cannot be reproduced

`flutter doctor -v
[√] Flutter (Channel beta, 3.7.0-1.3.pre, on Microsoft Windows [Version 10.0.22000.1335], locale en-US)
    • Flutter version 3.7.0-1.3.pre on channel beta at C:\DevTools\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 9b4416aaa7 (2 days ago), 2023-01-04 17:29:34 -0600
    • Engine revision 7b8906637f
    • Dart version 2.19.0 (build 2.19.0-444.3.beta)
    • DevTools version 2.20.0

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.3.2)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.3.32819.101
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[√] VS Code (version 1.74.2)
    • VS Code at C:\Users\John\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.56.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22000.1335]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 108.0.5359.125
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 108.0.1462.46

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!`

**Comment by domesticmouse** [comment]
Cool! 

I'd suggest you build your app on `beta` and raise bug reports against anyone whose plugins or packages don't work on beta. You'll be doing them a service in helping them get ready for next stable release 😎

**Comment by JohnF17** [comment]
> raise bug reports against anyone whose plugins or packages don't work on beta.

That's really good advice as fluent_ui and a few others don't yet seem to support the upcoming releases, and that's worrying to say the least 😥

> I'd suggest you build your app on `beta`

I'd really love to as it seems to solve this particular issue but the problem is with the direct dependencies (such as fluent_ui) and some transitive dependencies (such as sortedmap) that I rely heavily on with this current project that don't yet seem to work with both the `beta` and 'master' channels.
(Not taking away credit from anything or anyone here i know they are all working as hard as they can to make their products as great as they are and I bet they have a lot on their plates, just stating how it is 🙂, and I will make sure I let them know)

**Comment by domesticmouse** [comment]
There is two potential outcomes here. When `stable` is updated, either your dependencies are updated, or you'll need to find replacements. Raising bugs and seeing how the maintainer reacts will give you a hint as to which direction.

Even better is that you fork their github repo, upgrade it to work on `beta`, and then open a pull request with your changes. Because helping a maintainer in this way improves both your life and theirs.

**Comment by loic-sharma** [comment]
@domesticmouse Looping back on your original issue, I'm still unable to repro this and will need more information to debug your issue. I wrote up [this guide](https://gist.github.com/loic-sharma/248f28b3ba4664dde08da68f46b312bd) on how to debug the Windows engine, that might help get more information. Also, feel free to drop something on my calendar and we can pair program this together :)

**Comment by domesticmouse** [comment]
I'm no longer seeing this in Flutter 3.7. Closing.

**Comment by MRKDaGods** [comment]
I can reproduce this using a `CupertinoTextField` by typing anything then pressing **backspace**.
**Flutter (Channel stable, 3.7.3, on Microsoft Windows [Version 10.0.22621.1265], locale en-GB)**

---

E/flutter (24947): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: 'package:flutter/src/services/hardware_keyboard.dart': Failed assertion: line 432 pos 16: '_pressedKeys.containsKey(event.physicalKey)': A KeyUpEvent is dispatched, but the state shows that the physical key is not pressed. If this occurs in real application, please report this bug to Flutter. If this occurs in unit tests, please ensure that simulated events follow Flutter's event model as documented in `HardwareKeyboard`. This was the event: KeyUpEvent#79422(physicalKey: PhysicalKeyboardKey#e6f4b(usbHidUsage: "0x0007002a", debugName: "Backspace"), logicalKey: LogicalKeyboardKey#147d9(keyId: "0x100000008", keyLabel: "Backspace", debugName: "Backspace"), character: null, timeStamp: 0:01:08.471861, synthesized)
hardware_keyboard.dart:432
E/flutter (24947): #0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:51:61)
E/flutter (24947): #1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:40:5)
E/flutter (24947): #2      HardwareKeyboard._assertEventIsRegular.<anonymous closure>
hardware_keyboard.dart:432
E/flutter (24947): #3      HardwareKeyboard._assertEventIsRegular
hardware_keyboard.dart:443
E/flutter (24947): #4      HardwareKeyboard.handleKeyEvent
hardware_keyboard.dart:535
E/flutter (24947): #5      KeyEventManager.handleKeyData
hardware_keyboard.dart:881
E/flutter (24947): #6      PlatformDispatcher._keyDataListener.<anonymous closure>.<anonymous closure> (dart:ui/platform_dispatcher.dart:430:41)
E/flutter (24947): #7      _invoke1 (dart:ui/hooks.dart:164:13)
E/flutter (24947): #8      PlatformDispatcher._keyDataListener.<anonymous closure> (dart:ui/platform_dispatcher.dart:428:7)
E/flutter (24947): #9      _invoke2 (dart:ui/hooks.dart:183:13)
E/flutter (24947): #10     _ChannelCallbackRecord.invoke (dart:ui/channel_buffers.dart:40:5)
E/flutter (24947): #11     _Channel.push (dart:ui/channel_buffers.dart:130:31)
E/flutter (24947): #12     ChannelBuffers.push (dart:ui/channel_buffers.dart:326:17)
E/flutter (24947): #13     PlatformDispatcher._dispatchPlatformMessage (dart:ui/platform_dispatcher.dart:664:22)
E/flutter (24947): #14     _dispatchPlatformMessage (dart:ui/hooks.dart:86:31)

---

**Comment by domesticmouse** [comment]
@MRKDaGods can you please raise a new bug with a reproduction case? This is, unfortunately, a new bug in different code, with a different owner. Feel free to reference me in the bug text so I can get it in front of @cbracken for rectification =)

**Comment by domesticmouse** [comment]
> I can reproduce this using a `CupertinoTextField` by typing anything then pressing **backspace**.
> **Flutter (Channel stable, 3.7.3, on Microsoft Windows [Version 10.0.22621.1265], locale en-GB)**

Can you please add a minimal reproduction case?

**Comment by MRKDaGods** [comment]
PS: Running on an android emulator, API level 33

**Comment by domesticmouse** [comment]
I'm failing to replicate this issue, on Windows desktop, Chrome, and Android API 33. Here's the lightly updated code under test:

```dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() => runApp(const MainApp());

class MainApp extends StatefulWidget {
  const MainApp({super.key});

  @override
  State<MainApp> createState() => _MainAppState();
}

class _MainAppState extends State<MainApp> {
  late final TextEditingController ctlr;

  @override
  void initState() {
    super.initState();

    ctlr = TextEditingController();
    ctlr.addListener(() {
      debugPrint('Text: ${ctlr.text}');
    });
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Align(
          alignment: Alignment.center,
          child: ListView(
            shrinkWrap: true,
            children: [
              Padding(
                padding: const EdgeInsets.symmetric(horizontal: 8),
                child: CupertinoTextField(controller: ctlr),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
```

Here's the obligatory `flutter doctor` output:

```console
PS C:\Users\brett\Documents\flutter-projects\backspace_testing> flutter doctor --verbose
[✓] Flutter (Channel stable, 3.7.5, on Microsoft Windows [Version 10.0.22621.1265], locale en-AU)
    • Flutter version 3.7.5 on channel stable at C:\Users\brett\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c07f788888 (28 hours ago), 2023-02-22 17:52:33 -0600
    • Engine revision 0f359063c4
    • Dart version 2.19.2
    • DevTools version 2.20.1

[✓] Windows Version (Installed version of Windows is version 10 or higher)

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at C:\Users\brett\AppData\Local\Android\sdk
    • Platform android-33-ext4, build-tools 33.0.2
    • Java binary at: C:\Users\brett\Documents\jdk-18\bin\java.exe
    • Java version OpenJDK Runtime Environment (build 18+36-2087)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Build Tools 2019 16.11.22)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools
    • Visual Studio Build Tools 2019 version 16.11.33130.400
    • Windows 10 SDK version 10.0.22621.0

[!] Android Studio (version 2022.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] IntelliJ IDEA Community Edition (version 2022.1)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2022.1
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.75.1)
    • VS Code at C:\Users\brett\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.58.0

[✓] Connected device (4 available)
    • sdk gphone64 x86 64 (mobile) • emulator-5554 • android-x64    • Android 13 (API 33) (emulator)
    • Windows (desktop)            • windows       • windows-x64    • Microsoft Windows [Version 10.0.22621.1265]
    • Chrome (web)                 • chrome        • web-javascript • Google Chrome 110.0.5481.104
    • Edge (web)                   • edge          • web-javascript • Microsoft Edge 107.0.1418.42

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

**Comment by MRKDaGods** [comment]
I have tested the sample you have sent above, and it actually runs smoothly. However, as soon as I have commented the lines shown below, the issue is then reproducible.

![image](https://user-images.githubusercontent.com/25166537/221156115-87acb204-7812-44ed-b054-0f1a35aafa07.png)

By the way, the error is only reproducible once right after a startup/reload.
I have also tried adding another `CupertinoTextField` with a different text editing controller, but the error seems to only kick in once. Looks like a global issue of some sort.

Here is my flutter doctor output
```
C:\Users\mamar>flutter doctor --verbose
[√] Flutter (Channel stable, 3.7.3, on Microsoft Windows [Version 10.0.22621.1265], locale en-GB)
    • Flutter version 3.7.3 on channel stable at D:\src\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 9944297138 (2 weeks ago), 2023-02-08 15:46:04 -0800
    • Engine revision 248290d6d5
    • Dart version 2.19.2
    • DevTools version 2.20.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at C:\Users\mamar\AppData\Local\Android\sdk
    • Platform android-33-ext4, build-tools 33.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-9505619)
    • All Android licenses accepted.

[X] Chrome - develop for the web (Cannot find Chrome executable at .\Google\Chrome\Application\chrome.exe)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.4.5)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.4.33403.182
    • Windows 10 SDK version 10.0.22621.0

[√] Android Studio (version 2022.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-9505619)

[√] VS Code (version 1.75.1)
    • VS Code at C:\Users\mamar\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.58.0

[√] Connected device (3 available)
    • sdk gphone64 x86 64 (mobile) • emulator-5554 • android-x64    • Android 13 (API 33) (emulator)
    • Windows (desktop)            • windows       • windows-x64    • Microsoft Windows [Version 10.0.22621.1265]
    • Edge (web)                   • edge          • web-javascript • Microsoft Edge 110.0.1587.50

[√] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

**Comment by MRKDaGods** [comment]
> I'm failing to replicate this issue, on Windows desktop, Chrome, and Android API 33. Here's the lightly updated code under test:
> 
> ```dart
> import 'package:flutter/cupertino.dart';
> import 'package:flutter/material.dart';
> 
> void main() => runApp(const MainApp());
> 
> class MainApp extends StatefulWidget {
>   const MainApp({super.key});
> 
>   @override
>   State<MainApp> createState() => _MainAppState();
> }
> 
> class _MainAppState extends State<MainApp> {
>   late final TextEditingController ctlr;
> 
>   @override
>   void initState() {
>     super.initState();
> 
>     ctlr = TextEditingController();
>     ctlr.addListener(() {
>       debugPrint('Text: ${ctlr.text}');
>     });
>   }
> 
>   @override
>   Widget build(BuildContext context) {
>     return MaterialApp(
>       home: Scaffold(
>         body: Align(
>           alignment: Alignment.center,
>           child: ListView(
>             shrinkWrap: true,
>             children: [
>               Padding(
>                 padding: const EdgeInsets.symmetric(horizontal: 8),
>                 child: CupertinoTextField(controller: ctlr),
>               ),
>             ],
>           ),
>         ),
>       ),
>     );
>   }
> }
> ```
> 
> Here's the obligatory `flutter doctor` output:
> 
> ```
> PS C:\Users\brett\Documents\flutter-projects\backspace_testing> flutter doctor --verbose
> [✓] Flutter (Channel stable, 3.7.5, on Microsoft Windows [Version 10.0.22621.1265], locale en-AU)
>     • Flutter version 3.7.5 on channel stable at C:\Users\brett\flutter
>     • Upstream repository https://github.com/flutter/flutter.git
>     • Framework revision c07f788888 (28 hours ago), 2023-02-22 17:52:33 -0600
>     • Engine revision 0f359063c4
>     • Dart version 2.19.2
>     • DevTools version 2.20.1
> 
> [✓] Windows Version (Installed version of Windows is version 10 or higher)
> 
> [✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
>     • Android SDK at C:\Users\brett\AppData\Local\Android\sdk
>     • Platform android-33-ext4, build-tools 33.0.2
>     • Java binary at: C:\Users\brett\Documents\jdk-18\bin\java.exe
>     • Java version OpenJDK Runtime Environment (build 18+36-2087)
>     • All Android licenses accepted.
> 
> [✓] Chrome - develop for the web
>     • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe
> 
> [✓] Visual Studio - develop for Windows (Visual Studio Build Tools 2019 16.11.22)
>     • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools
>     • Visual Studio Build Tools 2019 version 16.11.33130.400
>     • Windows 10 SDK version 10.0.22621.0
> 
> [!] Android Studio (version 2022.1)
>     • Android Studio at C:\Program Files\Android\Android Studio
>     • Flutter plugin can be installed from:
>       🔨 https://plugins.jetbrains.com/plugin/9212-flutter
>     • Dart plugin can be installed from:
>       🔨 https://plugins.jetbrains.com/plugin/6351-dart
>     ✗ Unable to find bundled Java version.
>     • Try updating or re-installing Android Studio.
> 
> [✓] IntelliJ IDEA Community Edition (version 2022.1)
>     • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2022.1
>     • Flutter plugin can be installed from:
>       🔨 https://plugins.jetbrains.com/plugin/9212-flutter
>     • Dart plugin can be installed from:
>       🔨 https://plugins.jetbrains.com/plugin/6351-dart
> 
> [✓] VS Code (version 1.75.1)
>     • VS Code at C:\Users\brett\AppData\Local\Programs\Microsoft VS Code
>     • Flutter extension version 3.58.0
> 
> [✓] Connected device (4 available)
>     • sdk gphone64 x86 64 (mobile) • emulator-5554 • android-x64    • Android 13 (API 33) (emulator)
>     • Windows (desktop)            • windows       • windows-x64    • Microsoft Windows [Version 10.0.22621.1265]
>     • Chrome (web)                 • chrome        • web-javascript • Google Chrome 110.0.5481.104
>     • Edge (web)                   • edge          • web-javascript • Microsoft Edge 107.0.1418.42
> 
> [✓] HTTP Host Availability
>     • All required HTTP hosts are available
> 
> ! Doctor found issues in 1 category.
> ```

Oh well. After further testing, the issue is reproducible with your test code.

**Comment by domesticmouse** [comment]
I'm confused. What steps are you taking to trigger the issue?

**Comment by MRKDaGods** [comment]
> I'm confused. What steps are you taking to trigger the issue?

I just ran the app, and try to type in anything in the textbox. As soon as I hit backspace, the issue occurs.

**Comment by domesticmouse** [comment]
> > I'm confused. What steps are you taking to trigger the issue?
> 
> I just ran the app, and try to type in anything in the textbox. As soon as I hit backspace, the issue occurs.

Can you please update to Flutter 3.7.5 and re-try?

**Comment by MRKDaGods** [comment]
> > > I'm confused. What steps are you taking to trigger the issue?
> > 
> > 
> > I just ran the app, and try to type in anything in the textbox. As soon as I hit backspace, the issue occurs.
> 
> Can you please update to Flutter 3.7.5 and re-try?

Still reproducible! Although the occur did not occur when I first ran the app, it did occur when I did a hot reload.

**Comment by domesticmouse** [comment]
I'm failing to reproduce, even after deleting the `addListener` on the `TextEditingController` and triggering hot reload.

@loic-sharma how's your reproduction doc coming along. Is it in the `flutter/flutter` wiki yet?

**Comment by loic-sharma** [comment]
> @loic-sharma how's your reproduction doc coming along. Is it in the `flutter/flutter` wiki yet?

@domesticmouse: Yes, you can find the guide to debug a locally built engine here: https://github.com/flutter/flutter/wiki/Debugging-the-engine#debugging-windows-builds-with-visual-studio

**Comment by domesticmouse** [comment]
I'm closing this issue out. If anyone is still seeing this issue, please open a new issue with debugging info generated from building a debug engine: https://github.com/flutter/flutter/wiki/Debugging-the-engine#debugging-windows-builds-with-visual-studio
